### PR TITLE
[stdlib/cygwin] Modified SwiftGlibc for Cygwin and some fix

### PR DESCRIPTION
**Pull request by tinysun212**
#### Resolved bug number: N/A

---

Before merging this pull request to apple/swift repository:
- [ ] Test pull request on Swift continuous integration.

<details>
  <summary>

Triggering Swift CI</summary>

The swift-ci is triggered by writing a comment on this PR addressed to the GitHub user @swift-ci. Different tests will run depending on the specific comment that you use. The currently available comments are:

**Smoke Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please smoke test |
| OS X platform | @swift-ci Please smoke test OS X platform |
| Linux platform | @swift-ci Please smoke test Linux platform |

 **Validation Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please test |
| OS X platform | @swift-ci Please test OS X platform |
| Linux platform | @swift-ci Please test Linux platform |

Note: Only members of the Apple organization can trigger swift-ci.
</details>

'import SwiftGlibc' is now working in Cygwin.
Add workaround about USER_LABEL_PREFIX for Clang bug.

### CODE REVIEW

File: stdlib/public/Platform/CMakeLists.txt [hunk 1]
```diff
@@ -30,7 +30,7 @@ add_swift_library(${swift_platform_name} IS_SDK_OVERLAY
 
 set(glibc_modulemap_target_list)
 foreach(sdk ${SWIFT_SDKS})
-  if(${sdk} STREQUAL "LINUX" OR ${sdk} STREQUAL "FREEBSD")
+  if("${sdk}" STREQUAL "LINUX" OR "${sdk}" STREQUAL "FREEBSD" OR "${sdk}" STREQUAL "CYGWIN")
```
Comment (gribozavr) [inline stdlib/public/Platform/CMakeLists.txt]: Could you add line breaks after `OR`s for readability?

File: stdlib/public/Platform/Platform.swift [hunk 2]
```diff
@@ -109,6 +112,8 @@ public var errno : Int32 {
   set(val) {
 #if os(OSX) || os(iOS) || os(watchOS) || os(tvOS) || os(FreeBSD)
     return __error().pointee = val
+#elseif os(Windows)
+    return __errno().pointee = val
 #else
```
Comment (gribozavr) [inline stdlib/public/Platform/Platform.swift]: This is becoming hard to read now, could you change `#else` to `#elseif os(Linux)`?  (Two spots in this file.)

File: stdlib/public/Platform/Platform.swift [hunk 3]
```diff
@@ -337,7 +346,7 @@ public var SEM_FAILED: UnsafeMutablePointer<sem_t> {
 #if os(OSX) || os(iOS) || os(watchOS) || os(tvOS)
   // The value is ABI.  Value verified to be correct for OS X, iOS, watchOS, tvOS.
   return UnsafeMutablePointer<sem_t>(bitPattern: -1)
-#elseif os(Linux) || os(FreeBSD)
+#elseif os(Linux) || os(FreeBSD) || os(Windows)
   // The value is ABI.  Value verified to be correct on Glibc.
   return UnsafeMutablePointer<sem_t>(bitPattern: 0)
```
Comment (gribozavr) [inline stdlib/public/Platform/Platform.swift]: Did you verify the value?  Could you add a comment about the library and version where you verified it?

File: stdlib/public/Platform/Platform.swift [hunk 4]
```diff
@@ -311,8 +316,12 @@ public var SIG_DFL: sig_t? { return nil }
...
 public var SIG_HOLD: sig_t { return unsafeBitCast(5, to: sig_t.self) }
-#elseif os(Linux) || os(FreeBSD)
+#elseif os(Linux) || os(FreeBSD) || os(Windows)
+#if os(Windows)
+public typealias sighandler_t = _sig_func_ptr
+#else
 public typealias sighandler_t = __sighandler_t
+#endif
 
 public var SIG_DFL: sighandler_t? { return nil }
```
Comment (gribozavr) [inline stdlib/public/Platform/Platform.swift]: Did you verify the values below with the actual headers?

Comment (tinysun212) [inline stdlib/public/Platform/Platform.swift]: Yes, the values were same with Linux's one.  I'll change to add the comment about library and version.

**Comment by tinysun212** [comment]
Rebased to top of master. Any more reviews?

**Comment by gribozavr** [comment]
LGTM, thank you!

**Comment by gribozavr** [comment]
@swift-ci Please test and merge

**Comment by tinysun212** [comment]
Failed on ```test/DebugInfo/bbentry-location.swift```, it is now fixed in other commit.

**Pull request merged**